numba/core/types/containers.py

```diff
@@ -698,11 +710,27 @@ def unify(self, typingctx, other):
         if isinstance(other, DictType):
             if not other.is_precise():
                 return self
+            else:
+                ukey_type = self.key_type == other.key_type
+                uvalue_type = self.value_type == other.value_type
+                if ukey_type and uvalue_type:
+                    siv = self.initial_value
+                    oiv = other.initial_value
+                    siv_none = siv is None
+                    oiv_none = oiv is None
+                    if siv_none and oiv_none:
+                        return DictType(ukey_type, uvalue_type)
```

Comment:
`ukey_type`, `uvalue_type` are `bool`
and i think this branch should execute if `not siv_none and not oiv_none` and the initial value doesn't match

Reply:
agree, this a) has bugs and b) is left over from when I was trying to make it do something else! 965a19be8 fixes.